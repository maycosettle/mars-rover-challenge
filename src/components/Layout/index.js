export * from './PageLayout';
export * from './PaperLayout';
export * from './PhotoLayout';
