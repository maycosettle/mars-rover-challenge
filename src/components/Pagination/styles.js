import classNames from 'classnames';

const useStyles = () => {
  const a =
    'border-t-2 border-transparent pr-1 inline-flex items-center text-sm font-medium text-primary-500 hover:text-primary-700 hover:border-primary-300';
  const button = 'font-semibold';
  return {
    container: 'border-t border-gray-200 px-4 flex items-center justify-between sm:px-0',
    pageDown: {
      div: '-mt-px w-0 flex-1 flex',
      a: classNames(a, 'pr-1'),
      button,
    },
    numberContainer: 'hidden md:-mt-px md:flex',
    number:
      'cursor-pointer border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium',
    dots: 'border-transparent text-gray-500 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium',
    pageUp: {
      div: '-mt-px w-0 flex-1 flex justify-end',
      a: classNames(a, 'pl-1'),
      button,
    },
  };
};

export default useStyles;
