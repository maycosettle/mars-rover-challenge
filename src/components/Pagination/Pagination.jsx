import { Button } from 'components/Button';
import PropTypes from 'prop-types';

import useStyles from './styles';

const Pagination = ({ currentPage, onPageUp, onPageDown, onPageClick }) => {
  const styles = useStyles();

  return (
    <nav className={styles.container}>
      <div className={styles.pageDown.div}>
        <a className={styles.pageDown.a}>
          <Button
            onClick={onPageDown}
            transparent
            color="primary-600"
            className={styles.pageDown.button}
            name="Previous"
          />
        </a>
      </div>
      <div className="hidden md:-mt-px md:flex">
        <a onClick={onPageClick} className={styles.number}>
          1
        </a>
        <a onClick={onPageClick} className={styles.number} aria-current="page">
          2
        </a>
        <a onClick={onPageClick} className={styles.number}>
          3
        </a>
        <a onClick={onPageClick} className={styles.number}>
          8
        </a>
        <a onClick={onPageClick} className={styles.number}>
          9
        </a>
        <a onClick={onPageClick} className={styles.number}>
          10
        </a>
        <span className={styles.dots}>...</span>
        <a onClick={onPageClick} className={styles.number}>
          {currentPage >= 10 && currentPage + 1}
        </a>
      </div>
      <div className={styles.pageUp.div}>
        <a className={styles.pageUp.a}>
          <Button onClick={onPageUp} transparent color="primary-600" className={styles.pageUp.button} name="Next" />
        </a>
      </div>
    </nav>
  );
};

Pagination.propTypes = {
  pageCount: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  currentPage: PropTypes.number,
  showingFrom: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  showingTo: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  totalElements: PropTypes.number,
  onPageChange: PropTypes.func,
};

Pagination.defaultProps = {
  pageCount: 1,
  currentPage: 0,
  showingFrom: 0,
  showingTo: 10,
  totalElements: 10,
  onPageChange: () => {},
};

export default Pagination;
