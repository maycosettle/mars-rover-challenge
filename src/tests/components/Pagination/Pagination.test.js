import { shallow } from 'enzyme';
import { Pagination } from 'components/Pagination';

describe('tests on <Pagination />', () => {
  const wrapper = shallow(<Pagination />);

  test('Should render Pagination correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
