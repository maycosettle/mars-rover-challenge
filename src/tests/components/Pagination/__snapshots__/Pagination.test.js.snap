// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <Pagination /> Should render Pagination correctly 1`] = `
<nav
  className="border-t border-gray-200 px-4 flex items-center justify-between sm:px-0"
>
  <div
    className="-mt-px w-0 flex-1 flex"
  >
    <a
      className="border-t-2 border-transparent pr-1 inline-flex items-center text-sm font-medium text-primary-500 hover:text-primary-700 hover:border-primary-300 pr-1"
    >
      <Button
        bgColor="primary"
        className="font-semibold"
        color="primary-600"
        disabled={false}
        name="Previous"
        onClick={[Function]}
        rounded="full"
        transparent={true}
        type="button"
      />
    </a>
  </div>
  <div
    className="hidden md:-mt-px md:flex"
  >
    <a
      className="cursor-pointer border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
    >
      1
    </a>
    <a
      aria-current="page"
      className="cursor-pointer border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
    >
      2
    </a>
    <a
      className="cursor-pointer border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
    >
      3
    </a>
    <a
      className="cursor-pointer border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
    >
      8
    </a>
    <a
      className="cursor-pointer border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
    >
      9
    </a>
    <a
      className="cursor-pointer border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
    >
      10
    </a>
    <span
      className="border-transparent text-gray-500 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
    >
      ...
    </span>
    <a
      className="cursor-pointer border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
    />
  </div>
  <div
    className="-mt-px w-0 flex-1 flex justify-end"
  >
    <a
      className="border-t-2 border-transparent pr-1 inline-flex items-center text-sm font-medium text-primary-500 hover:text-primary-700 hover:border-primary-300 pl-1"
    >
      <Button
        bgColor="primary"
        className="font-semibold"
        color="primary-600"
        disabled={false}
        name="Next"
        onClick={[Function]}
        rounded="full"
        transparent={true}
        type="button"
      />
    </a>
  </div>
</nav>
`;
